Fix all "golint" warnings in top-level package.

juliusv opened the pull request:
This fixes all [golint](https://github.com/golang/lint) warnings in the top-level "influxdb" package:

    database.go:316:2: can probably use "var tagStrings []string" instead
    database.go:1077:6: exported type RetentionPolicies should have comment or be unexported
    server.go:337:1: exported method Server.StartSelfMonitoring should have comment or be unexported
    server.go:345:3: can probably use "var points []Point" instead
    server.go:433:2: can probably use "var groups []group" instead
    server.go:498:2: can probably use "var groups []group" instead
    server.go:1417:1: comment on exported method Server.CreateRetentionPolicyIfNotExists should be of the form "CreateRetentionPolicyIfNotExists ..."
    server.go:1899:1: exported method Server.DropMeasurement should have comment or be unexported
    server.go:2153:10: if block ends with a return statement, so drop this else and outdent its block
    server.go:2695:2: can probably use "var rows []*influxql.Row" instead
    server.go:2876:11: if block ends with a return statement, so drop this else and outdent its block
    server.go:3804:1: exported method Server.StartReportingLoop should have comment or be unexported
    tx.go:133:4: can probably use "var mappers []influxql.Mapper" instead
    tx.go:231:1: exported method LocalMapper.Open should have comment or be unexported
    tx.go:254:1: exported method LocalMapper.Close should have comment or be unexported
    tx.go:348:1: exported method LocalMapper.Next should have comment or be unexported

I don't know if there is any aspiration of getting the entire codebase lint-clean eventually, so feel free to ignore this if it causes more noise than benefit.

otoolep commented:
Thanks @juliusv -- have you signed the CLA? Once you do so, I can merge.

otoolep commented on tx.go:
```diff
@@ -228,6 +228,7 @@ type LocalMapper struct {
 	isRaw           bool                   // if the query is a non-aggregate query
 }
 
+// Open implements influxql.Mapper.
```
Can you simply write `Open opens the LocalMapper`? Similarly for `Close()`.

juliusv commented:
@otoolep Yep, just signed the CLA before the PR.

Fixed the comment and squashed (you might want to check whether the `LocalMapper.Next()` doc comment looks fine).

Just curious, are more of these cleanup changes generally interesting or would you rather want to avoid them? They are generally trivial to do, except when having to add docstrings to functions whose purpose is non-trivial to understand. Then again, those usually present interesting learning opportunities.

otoolep commented:
We're always interested in code clean-ups, of any detail. Your GoDoc string is close enough -- thanks.

juliusv commented:
Good to know, thanks!

corylanou commented:
If we get all of them cleaned up, we can add golint to our tool chain and ci build tools to keep it clean.  Much appreciated!
